simple/stats/runner.py

```diff
@@ -261,100 +270,111 @@ def _vertical_specs_dcids(self,
+                       output_file=output_file,
                        db=self.db,
                        reporter=reporter,
                        is_main_dc=is_main_dc)
 
-  def _create_importer(self, input_fh: FileHandler) -> Importer:
-    input_file = input_fh.basename()
+  def _create_importer(self, input_file: File) -> Importer:
     import_type = self.config.import_type(input_file)
-    debug_resolve_fh = self.process_dir_fh.make_file(
-        f"{constants.DEBUG_RESOLVE_FILE_NAME_PREFIX}_{input_file}")
-    reporter = self.reporter.import_file(input_file)
+    sanitized_path = input_file.full_path().replace("://",
+                                                    "_").replace("/", "_")
+    debug_resolve_file = self.process_dir.open_file(
+        f"{constants.DEBUG_RESOLVE_FILE_NAME_PREFIX}_{sanitized_path}")
+    reporter = self.reporter.get_file_reporter(input_file)
 
     if import_type == ImportType.OBSERVATIONS:
       input_file_format = self.config.format(input_file)
       if input_file_format == InputFileFormat.VARIABLE_PER_ROW:
-        return VariablePerRowImporter(input_fh=input_fh,
+        return VariablePerRowImporter(input_file=input_file,
                                       db=self.db,
                                       reporter=reporter,
                                       nodes=self.nodes)
-      return ObservationsImporter(input_fh=input_fh,
+      return ObservationsImporter(input_file=input_file,
                                   db=self.db,
-                                  debug_resolve_fh=debug_resolve_fh,
+                                  debug_resolve_file=debug_resolve_file,
                                   reporter=reporter,
                                   nodes=self.nodes)
 
     if import_type == ImportType.EVENTS:
-      return EventsImporter(input_fh=input_fh,
+      return EventsImporter(input_file=input_file,
                             db=self.db,
-                            debug_resolve_fh=debug_resolve_fh,
+                            debug_resolve_file=debug_resolve_file,
                             reporter=reporter,
                             nodes=self.nodes)
 
     if import_type == ImportType.ENTITIES:
-      return EntitiesImporter(input_fh=input_fh,
+      return EntitiesImporter(input_file=input_file,
                               db=self.db,
                               reporter=reporter,
                               nodes=self.nodes)
 
-    raise ValueError(f"Unsupported import type: {import_type} ({input_file})")
+    raise ValueError(
+        f"Unsupported import type: {import_type} ({input_file.full_path()})")
+
+
+def _check_not_overlapping(input_store: Store, output_store: Store):
+  input_path = input_store.full_path()
+  output_path = output_store.full_path()
+  if fspath.issamedir(input_path, output_path) or fspath.isparent(
+      input_path, output_path) or fspath.isparent(output_path, input_path):
+    raise ValueError(
```

Comment:
Is it possible to log a warning and ignore such directories (for input files)?

Reply:
IMO it's quicker for a user to fix a broken setup if we fail immediately rather than allowing an import to finish but with missing data. WDYT?

Reply:
SG. Since this is only when including subdirs, which is new functionality, we can go with this.

Let's document this requirement alongside the doc for the new parameter.